R/SelfControlledCohort.R

```diff
@@ -224,10 +224,11 @@ runSelfControlledCohort <- function(connectionDetails,
   }
 
   # Check if connection already open:
-  if (is.null(connectionDetails$conn())) {
-    conn <- DatabaseConnector::connect(connectionDetails)
+  if (inherits(connectionDetails$conn, "DatabaseConnectorDbiConnection")) {
```

Comment:
This will only work for connections to RSQLite databases like Eunomia. All other dbms are of type `DatabaseConnectorJdbcConnection`. 

Why not use
```r
if ("conn"  %in% names(connectionDetails)) {
```